src/BugsnagUnity/Payload/Error.cs

```diff
@@ -137,19 +137,44 @@ internal static Error FromSystemException(System.Exception exception, System.Dia
         {
             var errorClass = exception.GetType().Name;
             var stackFrames = new System.Diagnostics.StackTrace(exception, true).GetFrames();
+           
 
-            StackTraceLine[] lines = null;
-
-            if (stackFrames != null && stackFrames.Length > 0)
+            if (errorClass == ANDROID_JAVA_EXCEPTION_CLASS)
```

Comment:
The addition of Android-specific handling is a bit unexpected and wasn't called out in the PR description.

Is it worth a few comments to explain why this is required?

Reply:
There's a (brief) explanation around [line 193](https://github.com/bugsnag/bugsnag-unity/pull/532/files#diff-18e358b00cc5c047e63661436c8c943364cff776be9ea7f016248fe594038d81L193) - this is the duplicate bit that's getting consolidated

Reply:
I added another comment explaining the JVM exception in the main thread situation 👍